### PR TITLE
[IMP] account, l10n_be_invoice_bba: update invoice reference fields 

### PR DESCRIPTION
**Pull request by william-andre**
OPW 1920906

l10n_be_invoice_bba:
remove the l10n_be_structured_comm field
'random'/'date' references will not be longer used
new structured_invoice_number and structured_partner_ref reference types

account:
new rf_invoice_number and rf_partner_ref added as possible invoice reference generation

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by william-andre** [comment]
TODO need to take care of #29735

**Comment by william-andre** [comment]
I think the l10n_be mode for bba_partner_ref is really bad. I mean... max 999 invoices per partner?
Should I remove it or completely rework it? Or keep it as is?

File: addons/l10n_be_invoice_bba/models/account_invoice.py [hunk 1]
```diff
@@ -18,65 +18,38 @@ class AccountInvoice(models.Model):
...
         self.ensure_one()
-        algorithm = self.company_id.l10n_be_structured_comm
-        if algorithm == 'date':
-            date = fields.Date.from_string(fields.Date.today())
-            doy = date.strftime('%j')
-            year = date.strftime('%Y')
+        partner_ref = self.partner_id.ref
+        partner_ref_nr = re.sub('\D', '', partner_ref or '')
+        if (len(partner_ref_nr) < 3) or (len(partner_ref_nr) > 7):
+            raise UserError(_('The Customer should have an Internal Reference with min 3 and max 7 digits'
```
Comment (qdp-odoo) [inline addons/l10n_be_invoice_bba/models/account_invoice.py]: don't throw an error here

Comment (qdp-odoo) [reply addons/l10n_be_invoice_bba/models/account_invoice.py]: don't use a sequence but always the invoice id

File: addons/account/models/account.py [hunk 2]
```diff
@@ -449,6 +449,9 @@ def _get_bank_statements_available_sources(self):
         compute='_compute_refund_seq_number_next',
         inverse='_inverse_refund_seq_number_next')
 
+    invoice_reference_type = fields.Selection(string='Communication Type', required=True, selection=[('none', 'Free'), ('partner', 'Based on Partner'), ('invoice', 'Based on Invoice')], default='none', help='You can set here the default communication that will appear on customer invoices, once validated, to help the customer to refer to that particular invoice when making the payment.')
+    invoice_reference_model = fields.Selection(string='Communication Model', required=True, selection=[('odoo', 'Odoo'),('euro', 'European')], default='odoo', help="You can choose different models for each type of reference. The default one is the Odoo reference.")
```
Comment (qdp-odoo) [inline addons/account/models/account.py]: "Communication Standard" seems more appropriate to me, what do you think?

**Comment by qdp-odoo** [comment]
runbot is red, you might need to rebase

**Comment by qdp-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**